x-pack/plugins/maps/public/classes/styles/vector/components/style_prop_editor.tsx

```diff
@@ -27,9 +28,11 @@ export interface Props<StaticOptions, DynamicOptions> {
   defaultDynamicStyleOptions: DynamicOptions;
   disabled?: boolean;
   disabledBy?: VECTOR_STYLES;
+  customIcons?: CustomIcon[];
   fields: StyleField[];
   onDynamicStyleChange: (propertyName: VECTOR_STYLES, options: DynamicOptions) => void;
   onStaticStyleChange: (propertyName: VECTOR_STYLES, options: StaticOptions) => void;
+  onCustomIconsChange?: (customIcons: CustomIcon[]) => void;
```

Comment:
Type as `onCustomIconsChange: (customIcons: CustomIcon[]) => void;`

Reply:
Classes like VectorStyleColorEditor and VectorStyleSizeEditor inherit from StylePropEditor but don't use `onCustomIconsChange`.

Reply:
then should onCustomIconsChange be moved to VectorStyleIconEditor?

Reply:
The `updateCustomIcons` function from `map_actions` is [mapped as a Prop in the StyleSettings](https://github.com/elastic/kibana/pull/113144/files#diff-5a51f172a8829fd3ab59a3de6829593a96a1cf577c328828c86d66f9714d9a98R28-R30) and [passed to the `layer.renderStyleEditor` method](https://github.com/elastic/kibana/pull/113144/files#diff-55ace747cd41e863aac6d94a959a65658103432df0eec32f858141279b140f31R22-R23). I'm not sure if there's a better way to connect the map action directly to VectorStyleIconEditor?